File: tournaments/candles/candles.cs
int candles(int candlesNumber, int makeNew) {
    var result = 0;
    var a = 0;

    while (candlesNumber > 0) {
        result += candlesNumber;
        a += candlesNumber;
        candlesNumber = a / makeNew;
        a %= makeNew;
    }

    return result;
}

File: tournaments/candles/candles.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub candles {
    my ($candlesNumber, $makeNew) = @_;

    my $result = 0;
    my $temp = 0;

    while (0 < $candlesNumber) {
        $result += $candlesNumber;
        $temp += $candlesNumber;
        $candlesNumber = int $temp / $makeNew;
        $temp %= $makeNew;
    }

    return $result;
}
